packages/core/lib/console-child.js

```diff
@@ -0,0 +1,51 @@
+const Command = require("../lib/command");
+const TruffleError = require("@truffle/error");
+const Config = require("@truffle/config");
+const Web3 = require("web3");
+
+const inputStrings = process.argv[2];
+//deserialize options
+const options = JSON.parse(process.argv[3]);
+
+//deserialize networks entries
+for (var key in options.networks) {
+  options.networks[key] = JSON.parse(options.networks[key]);
+}
+
+//detect config so we can get the provider and resolver without having to serialize
+//and deserialize them
+const configDetected = Config.detect(options);
```

Comment:
What about naming this `detectedConfig`? `configDetected` sort of sounds like a predicate.

Reply:
Agree. Changed.